src/SSVOpenHexagon/Core/HGUpdate.cpp

```diff
@@ -25,14 +25,28 @@ void HexagonGame::update(ssvu::FT mFT)
 {
     mFT *= Config::getTimescale();
 
-    // TODO: refactor to avoid repetition, and truncate floating point number
-    // TODO: also show best record (here) and last run + best record (in menu)
-    steamManager.set_rich_presence_in_game(
-        levelData->name, status.getTimeSeconds());
+    // TODO: show best record (here) and last run + best record (in menu)
+
+    // Name formatter
+    std::string nameStr = std::string(levelData->name);
+    nameStr[0] = std::toupper(nameStr[0]);
+
+    // Difficulty multipler formatter
+    std::string diffStr = std::to_string(HexagonGame::difficultyMult);
+    size_t endPos = diffStr.find_last_not_of('0');
+    diffStr.erase(endPos + 1 + (int)(diffStr[endPos] == '.'), std::string::npos); // at least 1 dp
```

Comment:
I'm not sure what you're trying to achieve here, but it's very complicated. There must be a better way. What kind of formatting are you trying to get?

Reply:
This is so difficulty multipliers can be formatted to at least 1 decimal place.  For example, difficulty multipliers for Apeirogon could be displayed as [x0.35], [x0.5], [x0.75], [**x1.0**], [x1.25], [x1.4], [x1.6].

I was getting an issue where it was displaying as [**x1.**], and same for the time. I do feel it is a bit overcomplicated. Do ya have any suggestions on how I can improve this?

Reply:
Something like this?
https://gcc.godbolt.org/z/zWM8Ks